tests/cnf/core/network/metallb/tests/bgp-remote-as-dynamic.go

```diff
@@ -0,0 +1,215 @@
+					frrk8sPods, frrPod := setupBGPRemoteASTestCase(addressPool, hubIPv4ExternalAddresses, externalAdvertisedIPv4Routes,
+						externalAdvertisedIPv6Routes, dynamicASiBGP, tsparams.RemoteBGPASN)
+
+					By("Checking that BGP session is down")
+					verifyMetalLbBGPSessionsAreDownOnFrrPod(frrPod, removePrefixFromIPList(ipv4NodeAddrList))
+
+					By("Validating external FRR AS number received is incorrect and marked as 0 on the FRR nodes")
+					Eventually(func() error {
+						return frr.ValidateBGPRemoteAS(frrk8sPods, ipv4metalLbIPList[0], 0)
+					}, 60*time.Second, 5*time.Second).Should(Succeed(),
+						fmt.Sprintf("The remoteASN does not match the expected AS: %d", 0))
+				})
+		})
+	})
+
+func createBGPPeerWithDynamicASN(peerIP, dynamicASN string, eBgpMultiHop bool) {
+	By("Creating BGP Peer")
+
+	bgpPeer := metallb.NewBPGPeerBuilder(APIClient, "testpeer", NetConfig.MlbOperatorNamespace,
+		peerIP, tsparams.LocalBGPASN, 0).WithDynamicASN(mlbtypesv1beta2.DynamicASNMode(dynamicASN)).
+		WithPassword(tsparams.BGPPassword).WithEBGPMultiHop(eBgpMultiHop)
+
+	_, err := bgpPeer.Create()
+	Expect(err).ToNot(HaveOccurred(), "Failed to create BGP peer")
+}
+
+func deployFrrExternalPod(addressPool, hubIPAddresses, externalAdvertisedIPv4Routes,
+	externalAdvertisedIPv6Routes []string, localAS int) *pod.Builder {
+	By("Creating an IPAddressPool and BGPAdvertisement")
+
+	ipAddressPool := setupBgpAdvertisement(addressPool, int32(32))
+
+	By("Creating a MetalLB service")
+	setupMetalLbService(netparam.IPV4Family, ipAddressPool, "Cluster")
+
+	By("Creating nginx test pod on worker node")
+	setupNGNXPod(workerNodeList[0].Definition.Name)
+
+	By("Creating External NAD")
+	createExternalNad(tsparams.ExternalMacVlanNADName)
+
+	By("Creating static ip annotation")
+
+	staticIPAnnotation := pod.StaticIPAnnotation(
+		externalNad.Definition.Name, []string{fmt.Sprintf("%s/%s", ipv4metalLbIPList[0], "24")})
+
+	By("Creating MetalLb configMap")
+
+	masterConfigMap := createConfigMapWithStaticRoutes(localAS, ipv4NodeAddrList, hubIPAddresses,
+		externalAdvertisedIPv4Routes, externalAdvertisedIPv6Routes, false, false)
+
+	By("Creating FRR Pod")
+
+	frrPod := createFrrPod(
+		masterNodeList[0].Object.Name, masterConfigMap.Definition.Name, []string{}, staticIPAnnotation)
+
+	return frrPod
+}
+
+func setupBGPRemoteASTestCase(addressPool, hubIPv4ExternalAddresses, externalAdvertisedIPv4Routes,
+	externalAdvertisedIPv6Routes []string, dynamicAS string, externalFrrAS int) ([]*pod.Builder, *pod.Builder) {
+	var frrK8WebHookServer = "frr-k8s-webhook-server"
+
+	By("Creating a new instance of MetalLB Speakers on workers")
+
+	err := metallbenv.CreateNewMetalLbDaemonSetAndWaitUntilItsRunning(tsparams.DefaultTimeout, workerLabelMap)
+	Expect(err).ToNot(HaveOccurred(), "Failed to recreate metalLb daemonset")
+
+	By("Waiting until the new frr-k8s-webhook-server deployment is in Ready state.")
+
+	frrk8sWebhookDeployment, err := deployment.Pull(
```

Comment:
Why do we check frrk8sWebhookDeployment?

Reply:
Each time metallb is deployed it spins up a new frr-k8s-webhook-server. Just making sure it comes up as an indication that metallb is functioning correctly. Change it?

Reply:
Changed to daemonset on the frrk8 pods